### PR TITLE
migrate scala sdk extension impl from `intellij-bsp`

### PR DESCRIPTION
**Pull request by xuansontrinh**
what were done:

- register an additional synthetic scala library and convert it to a scala sdk by internal scala plugin API.

### CODE REVIEW

File: scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/ScalaSdkGetter.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.jetbrains.magicmetamodel.impl.workspacemodel.impl.updaters.transformers.{ScalaUtil => BspScalaUtil}
+import org.jetbrains.plugins.bsp.extension.points.ScalaSdkGetterExtension
+import org.jetbrains.plugins.bsp.server.tasks.ScalaSdk
+import org.jetbrains.plugins.scala.project.ScalaLanguageLevel
+import org.jetbrains.plugins.scala.project.external.ScalaSdkUtils
+
+import java.net.URI
+import java.nio.file.Paths
+
+class ScalaSdkGetter extends ScalaSdkGetterExtension {
```
Comment (unkarjedy) [inline scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/project/bsp/ScalaSdkGetter.scala]: I haven't seen the sources of BSP plugin, but the name `ScalaSdkGetterExtension` looks confusing for something which doesn't "gets scala sdk".
The logic seems to be about "registering Scala SDK" (in the module?) or "adding scala sdk to module".
Both "register" and "add" sound kinda opposite to "get"
I would suggest renaming it in the BSP plugin

**Comment by unkarjedy** [comment]
Note that you would need to duplicate the changes in idea241.x branch once you publish BSP plugin for 2024.1

**Comment by azdrojowa123** [comment]
as this comment above will be corrected LGTM

**Review by azdrojowa123 — APPROVED** [review]

**Comment by unkarjedy** [comment]
JFTR, the change will only be merged into idea233.x and won't be available in idea241.x automatically, because the BSP plugin is not published for IDEA 2024.1 yet

**Pull request merged**